### DISCUSSION TITLE
Is it possible to somehow define Slack request signature verfiying in Smithy?

### DISCUSSION THREAD

**Discussion by x1a0** [smithy-lang/smithy-rs#3275] [Q&A]
I am very new to Smithy and want to create a service that will receive requests from Slack.
Slack supports request verifying: https://api.slack.com/authentication/verifying-requests-from-slack.
In short, it needs 2 values from HTTP header, plus the whole HTTP body, as input, to calculate the signature.

I wonder is it possible to define this as some sort of Smithy trait. So I can have something like:
```
@slackSig
@http(method: "POST", uri: "/slack/event")
operation SlackEvent {
    input: SlackEventInput
}
```

Then the Rust code generator generates some Rust `trait` that I could implement, to have the actual Slack signature verifying logic. In the end, an instance of the verifier is created and attached to the service.

**Comment by david-perez** [comment]
smithy-rs doesn't have such a general mechanism to pre-apply a middleware based on a trait in the Smithy data model. We do plan on supporting such a mechanism, but the author of the Smithy trait will have to provide a small code-generation component: essentially, they will have to fill one of these out:

https://github.com/smithy-lang/smithy-rs/blob/f0929e74ba5859759c8cdf7b95e27d0b8b244f6e/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServiceConfigGenerator.kt#L25-L29

Iin your mind, how did you envision the code generator to generate such a Rust trait just from the `@slackSig` Smithy trait? Presumably you'll need to tell it somehow on which struct the trait should be implemented, what are its methods, etc?
